drivers/bmp180/bmp180.c

```diff
@@ -39,45 +42,42 @@ static int _compute_b5(bmp180_t *dev, int32_t ut, int32_t *b5);
  *                          BMP180 Core API                                 *
  *---------------------------------------------------------------------------*/
 
-int bmp180_init(bmp180_t *dev, i2c_t i2c, uint8_t mode)
+int bmp180_init(bmp180_t *dev, const bmp180_params_t *params)
 {
-    dev->i2c_dev = i2c;
+    dev->params = *params;
 
     /* Clamp oversampling mode */
-    if (mode > BMP180_ULTRAHIGHRES) {
-        mode = BMP180_ULTRAHIGHRES;
+    if (OVERSAMPLING > BMP180_ULTRAHIGHRES) {
+        OVERSAMPLING = BMP180_ULTRAHIGHRES;
     }
 
-    /* Setting oversampling mode */
-    dev->oversampling = mode;
-
     /* Initialize I2C interface */
-    if (i2c_init_master(dev->i2c_dev, I2C_SPEED_NORMAL)) {
+    if (i2c_init_master(DEV_I2C, I2C_SPEED_NORMAL)) {
         DEBUG("[Error] I2C device not enabled\n");
-        return -1;
+        return -BMP180_ERR_NOI2C;
     }
 
     /* Acquire exclusive access */
-    i2c_acquire(dev->i2c_dev);
+    i2c_acquire(DEV_I2C);
 
     /* Check sensor ID */
     uint8_t checkid;
-    i2c_read_reg(dev->i2c_dev, BMP180_ADDR, BMP180_REGISTER_ID, &checkid);
+    i2c_read_reg(DEV_I2C, BMP180_ADDR, BMP180_REGISTER_ID, &checkid);
```

Comment:
typically the address is also part of params, this way it would be hard coded to default address you choose in the header. Compare other drivers, e.g., your own rework of [BMX280](https://github.com/RIOT-OS/RIOT/blob/master/drivers/bmx280/include/bmx280_params.h#L39)